mailbox/api/src/test/java/org/apache/james/mailbox/MailboxManagerTest.java

```diff
@@ -558,7 +558,7 @@ void updateAnnotationsShouldThrowExceptionIfMailboxDoesNotExist() throws Excepti
             MailboxPath inbox = MailboxPath.inbox(session);
 
             assertThatThrownBy(() -> mailboxManager.updateAnnotations(inbox, session, ImmutableList.of(privateAnnotation)))
-                .isInstanceOf(MailboxNotFoundException.class);
+                .hasCauseInstanceOf(MailboxNotFoundException.class);
```

Comment:
I did not review everything but that's my first question to try to understand this changeset.

In this commit MailboxAnnotationManager did not changed and is still throwing "MailboxException". So why do you need this change? Did the API changed in a non documented way?

Reply:
Well that's likely something that will break and that I will need to revert.

Anyway you are right, this work is not ready for review, I'll draft it.